avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -662,10 +667,8 @@ private void processArray(JVar arraySchemaVar, final String name, final Schema a
       }
       doLoop.body().assign(chunkLen, JExpr.direct(DECODER + ".arrayNext()"));
     }, else1 -> {
-      if (useGenericTypes) {
+      if (finalAction.getShouldRead()) {
         else1.assign(arrayVar, finalNewArrayExp);
```

Comment:
this generates an arraylist of size 10, i think it should make it smaller (since its always empty here)